src/InternPoolEventSource.cs

```diff
@@ -42,14 +41,15 @@ private InternPoolEventSource() : base("InternPool")
 
         private void UpdateStats()
         {
-            var lastCheck = _lastCheck;
             var newCheck = Environment.TickCount64;
-            if (lastCheck == Interlocked.CompareExchange(ref _lastCheck, newCheck, lastCheck))
+            var lastUpdate = Volatile.Read(ref _lastUpdate);
+
+            if (newCheck - lastUpdate > 1000)
             {
-                if (newCheck - _lastUpdate > 1000)
+                if (lastUpdate == Interlocked.CompareExchange(ref _lastUpdate, newCheck, lastUpdate))
                 {
-                    _stats = _pool!.Stats;
-                    _lastUpdate = newCheck;
+                    Debug.Assert(_pool is not null, "EventSource should be enabled");
+                    _stats = _pool.Stats;
```

Comment:
I assume that `_stats` is written / read from the same thread? So visibility isn't a concern (on ARM)?